testutil/keeper/proof.go

```diff
@@ -45,50 +58,129 @@ func ProofKeeper(
 
 	ctrl := gomock.NewController(t)
 	mockSessionKeeper := mocks.NewMockSessionKeeper(ctrl)
-	mockSessionKeeper.EXPECT().
-		GetSession(gomock.Any(), gomock.Any()).
-		DoAndReturn(
-			func(
-				ctx context.Context,
-				req *sessiontypes.QueryGetSessionRequest,
-			) (*sessiontypes.QueryGetSessionResponse, error) {
-				session, ok := sessionByAppAddr[req.GetApplicationAddress()]
-				require.Truef(t, ok, "application address not provided during mock construction: %q", req.ApplicationAddress)
-
-				return &sessiontypes.QueryGetSessionResponse{
-					Session: &sessiontypes.Session{
-						Header: &sessiontypes.SessionHeader{
-							ApplicationAddress:      session.GetApplication().GetAddress(),
-							Service:                 req.GetService(),
-							SessionStartBlockHeight: 1,
-							SessionId:               session.GetSessionId(),
-							SessionEndBlockHeight:   5,
-						},
-						SessionId:           session.GetSessionId(),
-						SessionNumber:       1,
-						NumBlocksPerSession: session.GetNumBlocksPerSession(),
-						Application:         session.GetApplication(),
-						Suppliers:           session.GetSuppliers(),
-					},
-				}, nil
-			},
-		).AnyTimes()
+	mockAppKeeper := mocks.NewMockApplicationKeeper(ctrl)
+	mockAccountKeeper := mocks.NewMockAccountKeeper(ctrl)
 
 	k, err := keeper.NewKeeper(
 		cdc,
 		runtime.NewKVStoreService(storeKey),
 		log.NewNopLogger(),
 		authority.String(),
 		mockSessionKeeper,
-		nil,
-		nil,
+		mockAppKeeper,
+		mockAccountKeeper,
 	)
 	require.NoError(t, err)
 
 	ctx := sdk.NewContext(stateStore, cmtproto.Header{}, false, log.NewNopLogger())
 
-	// Initialize params
-	require.NoError(t, k.SetParams(ctx, types.DefaultParams()))
-
 	return k, ctx
 }
+
+func NewProofKeeperWithDeps(t testing.TB) (ProofKeeperWithDeps, context.Context) {
```

Comment:
Please add godocs, good opportunity to explain the difference from the function above